zign/oidc_api.py

```diff
@@ -0,0 +1,335 @@
+import logging
```

Comment:
It looks like a lot of this is duplicated code from the existing `api.py` file, @daniel40392 can you refactor?